### ISSUE TITLE
[deeplab v3+] I want to use the model on my own dataset, which just has 8 classes

### ISSUE THREAD

**Issue by smmdream** [tensorflow/models#4170]
### InvalidArgumentError (see above for traceback): Assign requires shapes of both tensors to match. lhs shape= [8] rhs shape= [21]
	 [[Node: save/Assign_113 = Assign[T=DT_FLOAT, _class=["loc:@logits/semantic/biases"], use_locking=true, validate_shape=true, _device="/job:localhost/replica:0/task:0/device:CPU:0"](logits/semantic/biases, save/RestoreV2_113)]]

The class number of my dataset is 8, and the model which trains on the VOC2012 has 21 classes,  I don't know how to train on my own dataset, could you help me?

**Comment by 95xueqian** [comment]
https://github.com/tensorflow/models/issues/3730   may help you

**Comment by smmdream** [comment]
@95xueqian  Thank you very much, I do as the answer say, but it doesn't solve my question, at last, I delete the file named checkpoint in ./datasets/segmentation_dataset/exp/train_on_trainval_set/train/ , then solve my problem. Thank you very much.

**Comment by shaunster0** [comment]
@smmdream thanks for this, I had this problem too and ur advice about /datasets/segmentation_dataset/exp/train_on_trainval_set/train/  worked

**Comment by kayvonkhosrowpour** [comment]
@smmdream What files specifically did you delete? I am having a similar error when I try to run a modified version of vis.py.

**Comment by beyou923** [comment]
> @95xueqian Thank you very much, I do as the answer say, but it doesn't solve my question, at last, I delete the file named checkpoint in ./datasets/segmentation_dataset/exp/train_on_trainval_set/train/ , then solve my problem. Thank you very much.

hello, I deleted the checkpoint file according to your method, run evel.py stuck here:
         INFO:tensorflow:Waiting for new checkpoint at ./datasets/ground_truth/exp/train_on_train_set/train
